Added new kb article handle-focus-unfocus-events-maui-chat (#1354)
telerik/maui-docs: knowledge-base/handle-focus-unfocus-events-maui-chat.md

---
title: Using Focus and Unfocus Events in Telerik UI for .NET MAUI Chat
description: Learn how to handle focus and unfocus events in Telerik UI for .NET MAUI Chat by accessing the RadMultilineTextInput control within the chat template.
type: how-to
page_title: Handle Focus and Unfocus Events in Telerik UI for .NET MAUI Chat
meta_title: Handle Focus and Unfocus Events in Telerik UI for .NET MAUI Chat
slug: handle-focus-unfocus-events-maui-chat
tags: telerik, ui for .net maui, chat, radmultilinetextinput, focus, unfocus
res_type: kb
ticketid: 1711957
---

## Environment

| Version | Product | Author | 
| --- | --- | ---- | 
| 13.0.0 | Telerik UI for .NET MAUI Chat | [Dobrinka Yordanova](https://www.telerik.com/blogs/author/dobrinka-yordanova) | 

## Description

> I want to handle the focus and unfocus events in the [Telerik UI for .NET MAUI Chat](https://www.telerik.com/maui-ui/documentation/controls/chart/overview) control. The `RadMultilineTextInput` is used as the input element within the Chat's `PromptInput`. How can I access the required events?

Reviewer: Use active voice whenever possible.
```suggestion
I want to handle the focus and unfocus events in the [Telerik UI for .NET MAUI Chat](https://www.telerik.com/maui-ui/documentation/controls/chart/overview) control. The Chat's `PromptInput` uses the `RadMultilineTextInput` as the input element. How can I access the required events?
```

This knowledge base article also answers the following questions:

- How to attach focus and unfocus events to the Chat input field in Telerik UI for .NET MAUI?
- How to handle focus events in Chat for .NET MAUI?
- How to subscribe to `RadMultilineTextInput` events in Telerik UI for .NET MAUI?

## Solution

> To handle the focus or unfocus event in Telerik UI for .NET MAUI Chat, access the `RadMultilineTextInput` inside the Chat's `PromptInput` on the `Loaded` event of the Chat control. Follow these steps:

Reviewer: ```suggestion
To handle the focus or unfocus event in Telerik UI for .NET MAUI Chat, access the `RadMultilineTextInput` inside the Chat's `PromptInput` on the `Loaded` event of the Chat control:
```

1. Subscribe to the `Loaded` event of the Chat control.
2. Access the visual tree descendants of the Chat to locate the `RadMultilineTextInput`.
3. Attach the `Focused` or `Unfocused` event to the `RadMultilineTextInput`.

> Here is an example implementation:

Reviewer: Aim at using less generic introductions.
```suggestion
The following code snippet shows how to handle the focus and unfocus events:
```

```csharp
private void Chat_Loaded(object sender, EventArgs e)
{

>     // Get all visual tree descendants of the chat control

Reviewer: Use a full stop at the end of each code comment as they are complete sentences.
